src/wasm/wasm-validator.cpp

```diff
@@ -2533,9 +2543,12 @@ void FunctionValidator::visitFunction(Function* curr) {
 
   shouldBeTrue(
     breakInfos.empty(), curr->body, "all named break targets must exist");
-  shouldBeTrue(exceptionTargetNames.empty(),
+  shouldBeTrue(delegateTargetNames.empty(),
                curr->body,
                "all named delegate targets must exist");
+  shouldBeTrue(rethrowTargetNames.empty(),
+               curr->body,
+               "all named rethrow targets must exist");
```

Comment:
Couldn't these only fail if there was a bug in the validator itself? If so, should these be asserts instead?

Reply:
Isn't that the same for all checking points within the validator? We already have similar lines:
https://github.com/WebAssembly/binaryen/blob/1d3f578d5dc26fc8ea83ea851ac0f1100a3cbad6/src/wasm/wasm-validator.cpp#L2534-L2538

Reply:
In other words, is it possible to construct an invalid Binaryen IR module that would trigger this error? (I would ask the same questions of any of the other checks we have)

Reply:
Yeah I think so. Some flawed pass can assign a `block` label to `delegate`, or even a random string (unlikely), but anyway I don't think it's impossible.

Reply:
But that would be caught by https://github.com/WebAssembly/binaryen/pull/3568/files#diff-2149be365b66a82038f1d3fa8f9fb1b4dcd40ab535c986f4bf0a4c37e669a1ceR2079-R2092, right? These checks are just checking that the target names were properly removed from their sets at the end of the corresponding scopes AFAICT, and it shouldn't be possible to create an invalid Binaryen module in which a try scope or catch scope never ends.

Reply:
Hmm, I think you're right. Then maybe I'll remove all those checks..?

Reply:
Removed.

Reply:
Oh no, you suggested to make them assertions... Will change them to assertions.